src/main/java/com/google/devtools/build/lib/remote/ByteStreamUploader.java

```diff
@@ -108,14 +113,16 @@ public ByteStreamUploader(
       ReferenceCountedChannel channel,
       @Nullable CallCredentials callCredentials,
       long callTimeoutSecs,
-      RemoteRetrier retrier) {
+      RemoteRetrier retrier,
+      ClientInterceptor... interceptors) {
```

Comment:
Did you consider passing in a channel that already has the interceptors attached instead? It seems preferable to me to not add knowledge about interceptors to the ByteStreamUploader if not necessary.

Reply:
Updated